processor-api/src/main/java/com/synopsys/integration/alert/processor/api/filter/DefaultFilterableNotificationExtractor.java

```diff
@@ -0,0 +1,76 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.processor.api.filter;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.synopsys.integration.alert.common.rest.model.AlertNotificationModel;
+import com.synopsys.integration.alert.processor.api.filter.extractor.NotificationExtractor;
+import com.synopsys.integration.alert.processor.api.filter.model.FilterableNotificationWrapper;
+import com.synopsys.integration.blackduck.api.manual.enumeration.NotificationType;
+
+@Component
+public class DefaultFilterableNotificationExtractor implements FilterableNotificationExtractor {
+    private final Logger logger = LoggerFactory.getLogger(DefaultFilterableNotificationExtractor.class);
+    private List<NotificationExtractor> notificationExtractors;
+
+    @Autowired
+    public DefaultFilterableNotificationExtractor(List<NotificationExtractor> notificationExtractors) {
+        this.notificationExtractors = notificationExtractors;
+    }
+
+    @Override
+    public final Optional<FilterableNotificationWrapper<?>> wrapNotification(AlertNotificationModel notification) {
+        String notificationTypeString = notification.getNotificationType();
+        NotificationType notificationType;
+        try {
+            notificationType = Enum.valueOf(NotificationType.class, notificationTypeString);
+        } catch (IllegalArgumentException e) {
+            logger.warn("Notification did not match any existing notification type: {}", notificationTypeString);
+            return Optional.empty();
+        }
+
+        Optional<NotificationExtractor> notificationExtractorOptional = getNotificationExtractor(notificationType);
+        if (notificationExtractorOptional.isPresent()) {
+            NotificationExtractor notificationExtractor = notificationExtractorOptional.get();
+
+            FilterableNotificationWrapper filterableNotificationWrapper = notificationExtractor.convertToFilterableNotificationWrapper(notification);
+            return Optional.of(filterableNotificationWrapper);
+        }
+
+        logger.warn("Did not find extractor for notification type: {}", notificationTypeString);
+        return Optional.empty();
+    }
+
+    private Optional<NotificationExtractor> getNotificationExtractor(NotificationType notificationType) {
+        return notificationExtractors.stream()
+                   .filter(notificationExtractor -> notificationExtractor.getNotificationType() == notificationType)
+                   .findFirst();
```

Comment:
Would people prefer i create a map for this so that I don't iterate over the extractors for every notification?

Reply:
That would be a great improvement!